apprise/plugins/NotifyNtfy.py

```diff
@@ -0,0 +1,264 @@
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+import requests
+import six
+from json import loads
+
+from .NotifyBase import NotifyBase
+from ..common import NotifyType
+from ..AppriseLocals import gettext_lazy as _
+
+
+# Priorities
+class NtfyPriority(object):
+    URGENT = 'urgent'
+    HIGH = 'high'
+    DEFAULT = 'default'
+    LOW = 'low'
+    MIN = 'min'
+
+    @classmethod
+    def get_value(cls, priority):
+        if priority == NtfyPriority.DEFAULT:  # most common case
+            return 3
+        elif priority == NtfyPriority.URGENT:
+            return 5
+        elif priority == NtfyPriority.HIGH:
+            return 4
+        elif priority == NtfyPriority.LOW:
+            return 2
+        elif priority == NtfyPriority.MIN:
+            return 1
+
+
+
+NTFY_PRIORITIES = (
+    NtfyPriority.URGENT,
+    NtfyPriority.HIGH,
+    NtfyPriority.DEFAULT,
+    NtfyPriority.LOW,
+    NtfyPriority.MIN,
+)
+
+
+class NotifyNtfy(NotifyBase):
+    """
+    A wrapper for Ntfy Notifications
+    """
+
+    # The default descriptive name associated with the Notification
+    service_name = 'Ntfy'
+
+    # The services URL
+    service_url = 'https://ntfy.sh/'
+
+    # Insecure protocol (for those self hosted requests)
+    protocol = 'ntfy'
+
+    # The default protocol
+    secure_protocol = 'ntfys'
+
+    # A URL that takes you to the setup/help of the specific protocol
+    setup_url = 'https://github.com/caronc/apprise/wiki/Notify_ntfy'
+
+    # Ntfy uses the https protocol
+    notify_url = 'https://ntfy.sh/{topic}'
+
+    # Message time to live (if remote client isn't around to receive it)
+    time_to_live = 2419200
+
+    # Define object templates
+    templates = (
+        '{schema}://{host}/{topic}',
+    )
+
+    # Define our template tokens
+    template_tokens = dict(NotifyBase.template_tokens, **{
+        'topic': {
+            'name': _('Topic'),
+            'type': 'string',
+            'required': False,
+        },
+        'host': {
```

Comment:
I have seen in other plugins that the port is split out into a separate token; not sure what the guideline is.